d2/src/main/java/com/linkedin/d2/xds/XdsToD2PropertiesAdaptor.java

```diff
@@ -219,13 +219,21 @@ public void onChanged(XdsClient.NodeUpdate update)
               _dualReadStateManager.reportData(serviceName, serviceProperties, true);
             }
           }
-          catch (PropertySerializationException e)
+          catch (Exception e)
           {
+            _serviceEventBus.publishInitialize(serviceName, null);
+            // still notify event bus to avoid timeout in case some subscribers are waiting for the data
             LOG.error("Failed to parse D2 service properties from xDS update. Service name: " + serviceName, e);
           }
         }
       }
 
+      @Override
+      public void onDelete()
+      {
+        _serviceEventBus.publishRemove(serviceName);
+      }
```

Comment:
As mentioned above, this can actually be removed altogether since it's handled elsewhere